### PR TITLE
CORE-5737: Remove membership-impl lib dependency in modules

### PR DESCRIPTION
**Pull request by YashNabar**
https://r3-cev.atlassian.net/browse/CORE-5737
Remove dependency on `membership-impl` lib where possible, remove related `package-info.java`, introduce `EndpointInfoFactory`.

### CODE REVIEW

File: libs/ledger/ledger-consensual-impl/src/main/kotlin/net/corda/ledger/consensual/impl/converter/PartyConverter.kt [hunk 1]
```diff
@@ -27,13 +27,15 @@ class PartyConverter @Activate constructor(
     override val type: Class<Party>
         get() = Party::class.java
 
-    override fun convert(context: ConversionContext): Party =
-        PartyImpl(
+    override fun convert(context: ConversionContext): Party {
+        println(context)
```
Comment (yift-r3) [inline libs/ledger/ledger-consensual-impl/src/main/kotlin/net/corda/ledger/consensual/impl/converter/PartyConverter.kt]: Remove the println

File: libs/membership/membership-common/src/main/kotlin/net/corda/membership/lib/EndpointInfoFactory.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+     * ```
+     *
+     * @param url The complete URL of the virtual node endpoint.
+     * @param protocolVersion The version of end-to-end authentication protocol. If multiple versions are supported,
+     * multiple instances of [EndpointInfo] can be created, each using a different protocol version. Defaults to
+     * [DEFAULT_PROTOCOL_VERSION].
+     */
+    fun create(
+        url: String,
+        protocolVersion: Int? = null
```
Comment (yift-r3) [inline libs/membership/membership-common/src/main/kotlin/net/corda/membership/lib/EndpointInfoFactory.kt]: Maybe `protocolVersion: Int = DEFAULT_PROTOCOL_VERSION`?

File: libs/ledger/ledger-consensual-impl/build.gradle [hunk 3]
```diff
@@ -12,7 +12,7 @@ dependencies {
 
     implementation platform("net.corda:corda-api:$cordaApiVersion")
     implementation project(":libs:membership:membership-common")
-    implementation project(':libs:membership:membership-impl')
+    implementation project(':libs:layered-property-map')
```
Comment (charlieR3) [inline libs/ledger/ledger-consensual-impl/build.gradle]: could this be a test implementation dependency?

File: components/membership/membership-client-impl/src/test/kotlin/net/corda/membership/impl/client/MGMOpsClientTest.kt [hunk 4]
```diff
@@ -93,9 +93,10 @@ class MGMOpsClientTest {
     private val knownKey: PublicKey = mock()
     private val keys = listOf(knownKey, knownKey)
 
+    private val endpointInfoFactory = EndpointInfoFactoryImpl()
```
Comment (charlieR3) [inline components/membership/membership-client-impl/src/test/kotlin/net/corda/membership/impl/client/MGMOpsClientTest.kt]: I noticed a couple test classes are using the actual implementation of `EndpointInfoFactoryImpl` even though they aren't testing that class. Would it be better to just use mocks and remove `impl` depenencies?

Comment (charlieR3) [reply components/membership/membership-client-impl/src/test/kotlin/net/corda/membership/impl/client/MGMOpsClientTest.kt]: I thought this wouldn't work after removing `package-info` also...

File: components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMember.kt [hunk 5]
```diff
@@ -9,14 +8,18 @@ import net.corda.membership.impl.registration.staticnetwork.StaticMemberTemplate
...
 import net.corda.utilities.time.UTCClock
 
 /**
  * Class which represents a static member. Static members are Map<String, Any> which we need
  * to map to Map<String, String>. This class provides access to properties always as strings.
  */
-class StaticMember(private val staticMemberData: Map<String, Any>) : Map<String, Any> by staticMemberData {
+class StaticMember(
+    private val staticMemberData: Map<String, Any>,
+    private val endpointInfoFactory: EndpointInfoFactory,
```
Comment (charlieR3) [inline components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMember.kt]: A lambda could be better here so we don't have essentially a data class holding a reference to a service class.
```suggestion
    private val endpointInfoFactory: (String, String) -> EndpointInfo,
```
(I might gotten the syntax wrong there, but something like that is what I mean)

**Review by charlieR3 — COMMENTED** [review]
Looks good - there's a lot of unit tests though using actual implementations where they could use mocks so that the test is only testing the classes it needs to test.

File: libs/ledger/ledger-consensual-impl/build.gradle (hunk 3 above)
Comment (YashNabar) [reply libs/ledger/ledger-consensual-impl/build.gradle]: It's also used in the `PartyConverter`

**Review by charlieR3 — APPROVED** [review]
LGTM

**Review by ronanbrowne — APPROVED** [review]
LGTM (build)

**Pull request merged**